ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/PaywallDataExtensions.kt

```diff
@@ -0,0 +1,100 @@
+}
+
+private fun PaywallData.Companion.createDefaultForIdentifiers(
+    packageIdentifiers: List<String>,
+    packageName: String,
+    currentColors: Colors,
+): PaywallData {
+    return PaywallData(
+        templateName = PaywallData.defaultTemplate.id,
+        config = PaywallData.Configuration(
+            packages = packageIdentifiers,
+            images = PaywallData.Configuration.Images(
+                background = PaywallData.defaultBackgroundImage,
+                icon = PaywallData.defaultAppIconPlaceholder,
+            ),
+            colors = PaywallData.defaultColors(currentColors),
+            blurredBackgroundImage = true,
+            displayRestorePurchases = true,
+        ),
+        localization = mapOf(Locale.US.toString() to PaywallData.defaultLocalization),
+        assetBaseURL = PaywallData.defaultTemplateBaseURL(packageName),
+        revision = PaywallData.revisionID,
+    )
+}
+
+// region Internal defaults
+
+internal val PaywallData.Companion.defaultTemplate: PaywallTemplate
+    get() = PaywallTemplate.TEMPLATE_2
+
+internal val PaywallData.Companion.defaultAppIconPlaceholder: String // TODO-PAYWALLS: use real icon
+    get() = "revenuecatui_default_paywall_app_icon"
+
+// endregion
+
+// region Private defaults
+
+private val PaywallData.Companion.revisionID: Int
+    get() = -1
+
+private val PaywallData.Companion.defaultBackgroundImage: String
+    get() = "default_background" // TODO-PAYWALLS: make sure this works
+
+private val PaywallData.Companion.defaultLocalization: PaywallData.LocalizedConfiguration
+    get() = PaywallData.LocalizedConfiguration(
+        title = "{{ app_name }}",
+        callToAction = "Continue",
+        offerDetails = "{{ total_price_and_per_month }}",
+        offerDetailsWithIntroOffer = "Start your {{ sub_offer_duration }} trial, then {{ total_price_and_per_month }}.",
+    )
+
+private fun PaywallData.Companion.defaultTemplateBaseURL(packageName: String): URL =
+    URL("android.resource://$packageName/")
+
+private fun PaywallData.Companion.defaultColors(currentColors: Colors): PaywallData.Configuration.ColorInformation {
+    return PaywallData.Configuration.ColorInformation(
+        light = getThemeColors(background = PaywallColor(colorInt = Color.White.toArgb()), currentColors),
+        dark = getThemeColors(background = PaywallColor(colorInt = Color.Black.toArgb()), currentColors),
+    )
+}
+
+// endregion
+
+private fun getThemeColors(
+    background: PaywallColor,
+    currentColors: Colors,
+): PaywallData.Configuration.Colors {
+    return PaywallData.Configuration.Colors(
+        background = background,
+        text1 = currentColors.primary.asPaywallColor(),
+        callToActionBackground = currentColors.secondary.asPaywallColor(),
+        callToActionForeground = background,
+        accent1 = currentColors.secondary.asPaywallColor(),
+        accent2 = currentColors.primary.asPaywallColor(),
+    )
+}
+
```

Comment:
For a future PR: we could add a preview in this file that shows `Template2()` with this default data :)

Reply:
great idea